target_chains/ethereum/contracts/forge-test/Pulse.t.sol

```diff
@@ -876,4 +876,207 @@ contract PulseTest is Test, PulseEvents {
         vm.prank(secondProvider);
         pulse.executeCallback(sequenceNumber, updateData, priceIds);
     }
+
+    function test_getFirstActiveRequests() public {
+        // Setup test data
+        (
+            bytes32[] memory priceIds,
+            bytes[] memory updateData
+        ) = setupTestData();
+        createTestRequests(priceIds);
+        completeRequests(updateData, priceIds);
+
+        testRequestScenarios(priceIds, updateData);
+    }
+
+    function setupTestData()
+        private
+        pure
+        returns (bytes32[] memory, bytes[] memory)
+    {
+        bytes32[] memory priceIds = new bytes32[](1);
+        priceIds[0] = bytes32(uint256(1));
+
+        bytes[] memory updateData = new bytes[](1);
+        return (priceIds, updateData);
+    }
+
+    function createTestRequests(bytes32[] memory priceIds) private {
+        uint256 publishTime = block.timestamp;
+        for (uint i = 0; i < 5; i++) {
+            vm.deal(address(this), 1 ether);
+            pulse.requestPriceUpdatesWithCallback{value: 1 ether}(
+                publishTime,
+                priceIds,
+                1000000
+            );
+        }
+    }
+
+    function completeRequests(
+        bytes[] memory updateData,
+        bytes32[] memory priceIds
+    ) private {
+        // Create mock price feeds and setup Pyth response
+        PythStructs.PriceFeed[] memory priceFeeds = createMockPriceFeeds(
+            block.timestamp
+        );
+        mockParsePriceFeedUpdates(priceFeeds);
+        updateData = createMockUpdateData(priceFeeds);
+
+        vm.deal(defaultProvider, 2 ether); // Increase ETH allocation to prevent OutOfFunds
+        vm.startPrank(defaultProvider);
+        pulse.executeCallback{value: 1 ether}(2, updateData, priceIds);
+        pulse.executeCallback{value: 1 ether}(4, updateData, priceIds);
+        vm.stopPrank();
+    }
+
+    function testRequestScenarios(
+        bytes32[] memory priceIds,
+        bytes[] memory updateData
+    ) private {
+        // Test 1: Request more than available
+        checkMoreThanAvailable();
+
+        // Test 2: Request exact number
+        checkExactNumber();
+
+        // Test 3: Request fewer than available
+        checkFewerThanAvailable();
+
+        // Test 4: Request zero
+        checkZeroRequest();
+
+        // Test 5: Clear all and check empty
+        clearAllRequests(updateData, priceIds);
+        checkEmptyState();
+    }
```

Comment:
not a strong opinion but if these are different tests maybe they can be different tests?